src/libtest/lib.rs

```diff
@@ -1084,12 +1084,16 @@ impl MetricMap {
 /// elimination.
 ///
 /// This function is a no-op, and does not even read from `dummy`.
+#[cfg(not(all(target_os = "nacl", target_arch = "le32")))]
 pub fn black_box<T>(dummy: T) -> T {
     // we need to "use" the argument in some way LLVM can't
     // introspect.
     unsafe {asm!("" : : "r"(&dummy))}
     dummy
 }
+#[cfg(all(target_os = "nacl", target_arch = "le32"))]
+#[inline(never)]
+pub fn black_box<T>(dummy: T) -> T { dummy }
```

Comment:
`black_box` should be `#[inline(always)]` fwiw (LLVM will inline the current definition virtually always anyway), the call cost is supposed to be missing there.

I didn't notice the `#[inline(never)]` in your case, I thought you just left it being a noop, but disabling inlining should work, except, well: there are corner cases: try leaving only one `black_box` call for any type you use it with.